diff --git a/test/functional/frontlinesms/legal/cases/DeleteCaseFromCaseDetailsPageSpec.groovy b/test/functional/frontlinesms/legal/cases/DeleteCaseFromCaseDetailsPageSpec.groovy
index 096fe2f..ce183ab 100644
--- a/test/functional/frontlinesms/legal/cases/DeleteCaseFromCaseDetailsPageSpec.groovy
+++ b/test/functional/frontlinesms/legal/cases/DeleteCaseFromCaseDetailsPageSpec.groovy
@@ -55,12 +55,10 @@ class DeleteCaseFromCaseDetailsPageSpec extends FrontlinesmsLegalGebSpec {
 
         when:
         deleteButton.click();
-        caseDeleteYes.click();
-        waitFor { !caseDeleteDialog.isVisible() }
+        deleteConfirmationDialog.confirm();
 
         then:
         assert at(SearchCasePage)
-
     }
 }
 
diff --git a/test/functional/frontlinesms/legal/functionaltests/pages/cases/ShowCasePage.groovy b/test/functional/frontlinesms/legal/functionaltests/pages/cases/ShowCasePage.groovy
index ca54f61..e430a0e 100644
--- a/test/functional/frontlinesms/legal/functionaltests/pages/cases/ShowCasePage.groovy
+++ b/test/functional/frontlinesms/legal/functionaltests/pages/cases/ShowCasePage.groovy
@@ -18,12 +18,10 @@ class ShowCasePage extends Page {
         statusMessage { $("div", id: "status").text() }
         errorMessage { $("div", id: "errorMessage").text() }
         linkedContactsRow {$(name:"contactRow", id: "contact-row")}
-        caseDeleteDialog { $("div", id: "caseDeleteDialog") }
-        caseDeleteYes { $("button", id: "confirm-yes")}
-        caseDeleteNo { $("button", id: "confirm-no")}
+        deleteConfirmationDialog { module ConfirmationDialog, $("div", id: "caseDeleteDialog").first() }
     }
-
 }
+
 class ContactRow extends Module {
     static content = {
         cell { i -> $("td", i) }
@@ -31,3 +29,15 @@ class ContactRow extends Module {
         primaryMobile { cell(1).text() }
     }
 }
+
+class ConfirmationDialog extends Module {
+    static content = {
+        yesButton { $("button", id: "confirm-yes")}
+        noButton { $("button", id: "confirm-no")}
+    }
+
+    def confirm = {
+        yesButton.click();
+        waitFor { !this.isVisible() }
+    }
+}
\ No newline at end of file
